service/history/workflow/update/update.go

```diff
@@ -494,6 +504,34 @@ func (u *Update) onResponseMsg(
 	return nil
 }
 
+// isIncomplete checks if update is already completed (rejected or processed).
+func (u *Update) isIncomplete() bool {
+	return !u.state.Matches(stateSet(stateProvisionallyCompleted | stateCompleted))
```

Comment:
@alexshtin was there a clear reason for including `stateProvisionallyCompleted` here? It can be rolled-back (is not durable) so the update is not necessarily complete. The reason I'm asking is that I need to skip reapply in the case where an update is in-progress, and I could use the `isIncomplete` method as the definition of in-progress. But I think that I would want to define `stateProvisionallyCompleted` as in-progress for that use case.